### PR TITLE
 bandtotridiagonal from GPU memory

### PR DESCRIPTION
**Pull request by rasolca**
Enabled the case where the matrix is on GPU for the GPU eigensolver.

Depends on #698.

### CODE REVIEW

**Comment by rasolca** [comment]
Note: it is based on #698 and currently the only relevant commit is the last one.

Depends on #698.

**Review by albestro — APPROVED** [review]
LGTM (just last commit https://github.com/eth-cscs/DLA-Future/pull/713/commits/7dad66279f4fbbda379f176c7d685ca3b276c88b, since it forks the other one)

File: include/dlaf/sender/traits.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ using SenderSingleValueType =
 template <typename Sender>
 using SenderElementType = typename SenderSingleValueType<Sender>::ElementType;
 
+// The value of an embedded device in the value_types of Sender, if it
+// exists and Sender sends exactly one type.
+template <typename Sender>
+inline constexpr Device SenderElementDevice = SenderSingleValueType<Sender>::device;
```
Comment (msimberg) [inline include/dlaf/sender/traits.h:58]:
```suggestion
inline constexpr Device SenderDevice = SenderSingleValueType<Sender>::device;
```
?

**Review by msimberg — APPROVED** [review]
Only a potential rename suggestion, otherwise this looks good to me.

**Comment by rasolca** [comment]
bors try

**Review by msimberg — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Pull request merged**